src/types/api.ts

```diff
@@ -134,6 +134,34 @@ export type PageSelector = {
   label?: string;
 };
 
+export type RectShape = {
+  type: 'rect';
+  left: number;
+  top: number;
+  bottom: number;
+  right: number;
+};
+
+export type PointShape = {
+  type: 'point';
+  x: number;
+  y: number;
+};
+
+export type Shape = RectShape | PointShape;
+
+/**
+ * Selector which identifies a region of the document defined by a shape.
+ *
+ * TODO:
+ *  - Specify the origin used by the shape coordinates (eg. page, document body)
+ *  - Specify the coordinate system
+ */
```

Comment:
For PDFs the origin and coordinate system in this PR are the ones I expect us to use in the final version. These TODOs are about how to make these selectors generalize to other document types. In HTML documents for example, a shape drawn on an `<img>` might be relative to the top-left corner of the image with a CSS selector specified as an anchor to identify the image within the document.